sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -32,138 +32,60 @@
 
 from azure.core.pipeline.policies import HttpLoggingPolicy
 
+from .http_constants import HttpHeaders
 
 
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs) -> None: # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
         if self._enable_diagnostics_logging:
-            self.logger = logger
-
-    def on_request(self, request): # pylint: disable=too-many-return-statements, too-many-statements
-        # type: (PipelineRequest) -> None
-        http_request = request.http_request
-        options = request.context.options
-        self._enable_diagnostics_logging = request.context.setdefault(
-            "enable_diagnostics_logging",
-            options.pop("enable_diagnostics_logging", self._enable_diagnostics_logging))
-        if self._enable_diagnostics_logging:
-            # Get logger in my context first (request has been retried)
-            # then read from kwargs (pop if that's the case)
-            # then use my instance logger
-            logger = request.context.setdefault("logger", options.pop("logger", self.logger))
-
+            cosmos_disallow_list = ["Authorization"]
+            cosmos_allow_list = [
+                v for k, v in HttpHeaders.__dict__.items() if not k.startswith("_") and k not in cosmos_disallow_list
+            ]
+            self.allowed_header_names.update(set(cosmos_allow_list))
+
+    def _format_error(self, payload: str) -> str:
+        payload = payload.replace("true", "True")
+        payload = payload.replace("false", "False")
+        output = ast.literal_eval(payload)
+        return output['message'].replace("\r", " ")
+
+    def on_response(
```

Comment:
I've removed `on_request` entirely - it looks like the only additions were the extra headers, which are already handled by adding them to the allow list.

Reply:
Ohh I see, since we overload the replace method with our own allowlist then we just use the remaining logic of the original HttpLoggingPolicy right?

Reply:
No we don't even need to overload the replace method - anything that's set up in the allow_list at construction will be logged by the base policy.